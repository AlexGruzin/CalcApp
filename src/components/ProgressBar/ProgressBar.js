import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './ProgressBar.styles.less';

const ProgressBar = ({ className, percent, label }) => {
  const [controlledPercent, setControlledPercent] = useState(percent);

  useEffect(() => {
    setControlledPercent(percent);
  }, [percent]);

  return (
    <div className={classNames('progress-bar__border', className)}>
      <div className="progress-bar__wrapper" style={{ width: `${controlledPercent}%` }}>
        <h6 className="progress-bar__label">{label}</h6>
        <div className="progress-bar__bar" />
      </div>
    </div>
  );
};

ProgressBar.propTypes = {
  label: PropTypes.string,
  className: PropTypes.string,
  percent: PropTypes.number,
};

ProgressBar.defaultProps = {
  label: '',
  className: '',
  percent: 0,
};

export default ProgressBar;
