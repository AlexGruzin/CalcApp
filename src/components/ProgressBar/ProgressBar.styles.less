.progress-bar__border {
  width: 300px;
}

.progress-bar__wrapper {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  transition: all 2s ease-in-out;
}

.progress-bar__bar {
  display: flex;
  width: 100%;
  min-height: 14px;
  height: 24px;
  transition: all 2s ease-in-out;
}

.progress-bar__label {
  margin: 0;
}
