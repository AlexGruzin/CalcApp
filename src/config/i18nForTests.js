import i18n from 'i18next';

// locales
import US from '../locales/en';

i18n.init({
  debug: false,
  fallbackLng: ['US'],
  lng: 'US',

  ns: ['auth', 'resetPassword'],
  defaultNS: 'auth',

  resources: {
    US,
  },
});

export default i18n;

