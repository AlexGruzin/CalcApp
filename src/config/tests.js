import React from 'react';
import { render } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next'
import i18n from "./i18nForTests";
import jsdom from 'jsdom';
const { JSDOM } = jsdom;

/*
 * jest-dom
 * https://github.com/testing-library/jest-dom
 */
import '@testing-library/jest-dom/extend-expect';

/*
 * jest-fetch-mock
 * https://github.com/jefflau/jest-fetch-mock
 */
require('jest-fetch-mock').enableMocks();

if (global.document) {
  global.document.createRange = () => ({
    setStart: () => {
    },
    setEnd: () => {
    },
    commonAncestorContainer: {
      nodeName: 'BODY',
      ownerDocument: document,
    },
    createContextualFragment: str => JSDOM.fragment(str),
    // createContextualFragment: str => document.createElement('hr'),
  });
  global.open = () => undefined;
}

export const renderWithi18next = Component => {
  const Comp = React.cloneElement(Component, {
    changeLanguage: lng => {
      i18n.changeLanguage(lng);
      rerender(<I18nextProvider i18n={i18n}>{Comp}</I18nextProvider>);
    }
  });
  const defaultRender = render(
    <I18nextProvider i18n={i18n}>{Component}</I18nextProvider>
  );
  const { rerender } = defaultRender;
  return defaultRender;
};


// /**
//  * Wraps a component with an Intl Provider.
//  * @param {Object} component The component to render
//  */
// const withTestIntl = component => <IntlProvider locale="en">{component}</IntlProvider>;
//
// /*
//  * https://testing-library.com/docs/example-react-intl
//  */
// export const renderWithReactIntl = component => render(withTestIntl(component));

