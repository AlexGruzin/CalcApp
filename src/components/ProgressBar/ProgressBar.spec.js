import React from 'react';
import { render } from '@testing-library/react';
import { renderWithi18next } from 'config/tests';

import ProgressBar from './ProgressBar';

describe('CampaignsTable tests', () => {
  it('Check snapshots of CampaignsTable are equal.', () => {
    const { asFragment } = renderWithi18next(<ProgressBar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
