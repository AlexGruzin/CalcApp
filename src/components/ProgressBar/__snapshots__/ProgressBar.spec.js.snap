// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignsTable tests Check snapshots of CampaignsTable are equal. 1`] = `
<DocumentFragment>
  <div
    class="progress-bar__border"
  >
    <div
      class="progress-bar__wrapper"
      style="width: 0%;"
    >
      <h6
        class="progress-bar__label"
      />
      <div
        class="progress-bar__bar"
      />
    </div>
  </div>
</DocumentFragment>
`;
